packages/checkout/widgets-lib/src/widgets/sale/functions/smartCheckoutUtils.ts

```diff
@@ -120,32 +135,60 @@ export const smartCheckoutTokensList = (
   return tokenSymbols;
 };
 
-export const filterSmartCheckoutResult = (smartCheckoutResult: SmartCheckoutResult): SmartCheckoutResult => {
+export const filterSmartCheckoutResult = (
+  smartCheckoutResult: SmartCheckoutResult,
+  provider?: Web3Provider,
+): SmartCheckoutResult => {
+  // if the transaction is sufficient or there are no routes found stays as is
   if (smartCheckoutResult.sufficient
     || smartCheckoutResult.router.routingOutcome.type !== RoutingOutcomeType.ROUTES_FOUND) {
     return smartCheckoutResult;
   }
 
+  // if passport wallet and only native balance is insufficient,
+  // return sufficient as passport transactions are gas sponsored
+  const isPassport = !!(provider?.provider as any)?.isPassport;
+  const onlyNativeBalanceIsInsufficient = smartCheckoutResult.transactionRequirements.every(
+    (req) => (req.type === ItemType.NATIVE ? !req.sufficient : req.sufficient),
+  );
+
+  if (isPassport && !smartCheckoutResult.sufficient && onlyNativeBalanceIsInsufficient) {
+    return {
+      sufficient: true,
+      transactionRequirements: smartCheckoutResult.transactionRequirements,
+    } as SmartCheckoutSufficient;
+  }
+
+  // otherwise, filter out disabled steps if required
+  const stepTypesToFiler = [
```

Comment:
at the moment of this PR, it has not been decided to remove BRIDGE as a secondary flow for a primary sale.

We are deferring deprecating this filter until its UX is clarified. Also to keep in mind, smart checkouts will at some point include a way to skip gas check or disable funding routes from configurations. When that happens this filtering will not longer required.

Reply:
cc @dreamoftrees